compiler/il/OMRSymbol.hpp

```diff
@@ -398,8 +393,32 @@ class OMR_EXTENSIBLE Symbol
    inline void setMemoryTypeShadowSymbol();
    inline bool isMemoryTypeShadowSymbol();
 
-   void setOrdered() { _flags.set(Ordered); }
-   bool isOrdered()  { return _flags.testAny(Ordered); }
+   enum MemoryOrdering
+      {
+      TransparentSemantics,
+      OptOpaqueSemantics,
+      AcqRelSemantics,
+      VolatileSemantics,
+
+      LastMemoryOrdering = VolatileSemantics
+      };
+
+   static inline const char *getMemoryOrderingName(MemoryOrdering ordering);
+
+   inline void setMemoryOrdering(MemoryOrdering ordering);
+   inline MemoryOrdering getMemoryOrdering();
+
+   inline void setTransparent();
+   inline bool isTransparent();
+
+   inline void setOptOpaque();
+   inline bool isOptOpaque();
+
+   inline void setAcqRel();
+   inline bool isAcqRel();
+
+   inline void setVolatile();
+   inline bool isVolatile();
```

Comment:
Please add Doxygen comments describing the semantics associated with each of these, or at least relevant references.